pydocstringformatter/_formatting/base.py

```diff
@@ -202,3 +206,70 @@ def treat_summary(
 
     def treat_description(self, description: str, indent_length: int) -> str:
         return description
+
+
+def _pairwise(iterator: Iterator[_T]) -> Iterator[tuple[_T, _T]]:
+    """Create an iterator over pairs of successive elements."""
+    first, second = tee(iterator)
+    next(second)
+    return zip(first, second)
+
+
+class NumpydocSectionFormatter(StringAndQuotesFormatter, metaclass=abc.ABCMeta):
+    """Base class for formatters working on numpydoc sections."""
+
+    style = ["numpydoc"]
+
+    @abc.abstractmethod
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Process the individual numpydoc sections."""
+
+    def treat_string(
+        self,
+        tokeninfo: tokenize.TokenInfo,
+        indent_length: int,
+        quotes: str,
+        quotes_length: Literal[1, 3],
+    ) -> str:
+        """Split numpydoc sections, pass them for processing, then rejoin them."""
+        # Split sections
+        lines = tokeninfo.string[quotes_length:-quotes_length].split("\n")
+        # handle the spaces before the closing quotes
+        if lines[-1].isspace():
+            lines[-1] = ""
+
+        section_heading_lines = [
+            index
+            for index, line in enumerate(lines)
+            if "-" in line and all(char in " \t-" for char in line)
+        ]
+        section_starts = (
+            [0] + [index - 1 for index in section_heading_lines] + [len(lines)]
+        )
+
+        sections = OrderedDict(
+            [
+                (
+                    lines[curr_section_start].lstrip(),
+                    lines[curr_section_start:next_section_start],
+                )
+                for curr_section_start, next_section_start in _pairwise(
+                    iter(section_starts)
+                )
+            ]
+        )
+        # Process sections
+        new_sections = self.treat_sections(sections)
+
+        # Rejoin sections
+        new_lines = [line for section in new_sections.values() for line in section]
+        # Ensure the last line puts the quotes in the right spot
+        # Enforces indented closing quotes on the last line
```

Comment:
This probably shouldn't be handled by this formatter. Can we store whether the closing quotes were already on a new line and do this according to that?

We like to keep each formatter responsible for a single thing to provide optimal customisation for users.

Reply:
So, new test for that.  Would it help if I made all the test input files symlinks of `numpydoc_style.py`, with only the changes for that formatter reflected in the corresponding `.py.out` file?

Reply:
That might work well in this case yeah!

Reply:
Whole bunch of symlinks created and `.args` files expanded